### PR TITLE
DBZ-2318 add column check constraints and JSON validation check constraints to debezium-ddl-parser ANTLR v4 MySQL grammar

### PR DESCRIPTION
**Pull request by rk3rn3r**
fixes https://issues.redhat.com/browse/DBZ-2318

### CODE REVIEW

**Comment by jpechane** [comment]
@rk3rn3r Nice fix, I've not noticed the check contraint is also not supported. While at it could you please add the support for *all* JSON related functions?

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4 [hunk 1]
```diff
@@ -2574,5 +2576,5 @@ functionNameBase
     | VALIDATE_PASSWORD_STRENGTH | VERSION | VISIBLE
     | WAIT_UNTIL_SQL_THREAD_AFTER_GTIDS | WEEK | WEEKDAY
     | WEEKOFYEAR | WEIGHT_STRING | WITHIN | YEAR | YEARWEEK
-    | Y_FUNCTION | X_FUNCTION
+    | Y_FUNCTION | X_FUNCTION | JSON_VALID | JSON_SCHEMA_VALID
```
Comment (gunnarmorling) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4:2579]: If there's something MariaDB-specific, could you please mark it with a comment (there's some examples where we already do this).

File: debezium-ddl-parser/src/test/resources/mysql/examples/ddl_create.sql [hunk 2]
```diff
@@ -184,3 +184,32 @@ BEGIN
     RETURN result;
 END;
 #end
+#begin
+-- From MariaDB 10.4.3, the JSON_VALID function is automatically used as a CHECK constraint for the JSON data type alias in order to ensure that a valid json document is inserted.
```
Comment (rk3rn3r) [inline debezium-ddl-parser/src/test/resources/mysql/examples/ddl_create.sql:188]: here is one of the comment you were looking for. I think they best fit here. WDYT @gunnarmorling ?

File: debezium-ddl-parser/src/test/resources/mysql/examples/ddl_create.sql [hunk 3]
```diff
@@ -184,3 +184,32 @@ BEGIN
...
+-- src: https://mariadb.com/kb/en/json_valid/
+CREATE TABLE `global_priv` (
+    `Host` CHAR(60) COLLATE utf8_bin NOT NULL DEFAULT '',
+    `User` CHAR(80) COLLATE utf8_bin NOT NULL DEFAULT '',
+    `Privilege` LONGTEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_bin NOT NULL DEFAULT '{}' CHECK (json_valid(`Privilege`)),
+    PRIMARY KEY (`Host`,`User`)
+) ENGINE=Aria DEFAULT CHARSET=utf8 COLLATE=utf8_bin PAGE_CHECKSUM=1 TRANSACTIONAL=1 COMMENT='Users and global privileges';
+#end
+#begin
+-- https://dev.mysql.com/doc/refman/8.0/en/json-validation-functions.html#json-validation-functions-constraints
```
Comment (rk3rn3r) [inline debezium-ddl-parser/src/test/resources/mysql/examples/ddl_create.sql:198]: and the other one @gunnarmorling.

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4 (hunk 1 above)
Comment (rk3rn3r) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4:2579]: @gunnarmorling comments are already in the unit tests. I thought in comparison its imo the best place to put the comment. WDYT?

**Comment by rk3rn3r** [comment]
> @rk3rn3r Nice fix, I've not noticed the check contraint is also not supported. While at it could you please add the support for _all_ JSON related functions?

@jpechane no, I don't think that's not neccessary. We only need support for that functions that are supported in DDL statements. I think that's only JSON_VALID for MariaDB and JSON_SCHEMA_VALID for MySQL. That's what I found out and what I did.

**Comment by jpechane** [comment]
@rk3rn3r We are consuming upstream project so I think it is fair to add missing support even for functions we don't directly use right now as others can benefit for them - and it wouldn't make sense to add those functions upstream and diverge on them in our grammar.

**Comment by jpechane** [comment]
Applied, thank @rk3rn3r

**Comment by gunnarmorling** [comment]
> We only need support for that functions that are supported in DDL statements. 

@rk3rn3r, the problem is that DDL can contain any kind of DML, too, e.g. in the declaration of a stored procedure. So we really need to be able to parse anything. So if there's something left, could you please log a follow-up Jira issue? Thanks!

**Pull request merged**